src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -46,7 +46,7 @@ public abstract partial class Frame : FrameContext, IFrameControl
 
         private readonly object _onStartingSync = new Object();
         private readonly object _onCompletedSync = new Object();
-        protected readonly FrameRequestHeaders _requestHeaders = new FrameRequestHeaders();
+        internal readonly FrameRequestHeaders _requestHeaders = new FrameRequestHeaders();
```

Comment:
So the chunked trailers can use the static method `while (!Frame.TakeMessageHeaders(input, frame._requestHeaders))`